centrallix-sysdoc/OSDriver_Authoring.md

```diff
@@ -115,497 +170,1032 @@ License:  Copyright (C) 2001-2011 LightSys Technology Services. See LICENSE.txt
+| --------------------------------------------------------- | ------------
+| [Open](#function-open)*                                   | Opens a new driver instance object on a given root node.
+| [OpenChild](#function-openchild)                          | ???
+| [Close](#function-close)*                                 | Close an open object created by either `Open()` or `QueryFetch()`.
+| [Create](#function-create)                                | Create a new driver root node object.
+| [Delete](#function-delete)                                | Delete an existing driver root node object.
+| [DeleteObj](#function-deleteobj)*                         | ???
+| [OpenQuery](#function-openquery)**                        | Start a new query for child objects of a given object.
+| [QueryDelete](#function-querydelete)                      | Delete specific objects from a query's result set.
+| [QueryFetch](#function-queryfetch)**                      | Open the next child object in the query's result set.
+| [QueryCreate](#function-querycreate)                      | ???
+| [QueryClose](#function-queryclose)**                      | Close an open query.
+| [Read](#function-read)*                                   | Read content from the object.
+| [Write](#function-write)*                                 | Write content to the object.
+| [GetAttrType](#function-getattrtype)*                     | Get the type of a given object's attribute.
+| [GetAttrValue](#function-getattrvalue)*                   | Get the value of a given object's attribute.
+| [GetFirstAttr](#function-getfirstattr--getnextattr)*      | Get the name of the object's first attribute.
+| [GetNextAttr](#function-getfirstattr--getnextattr)*       | Get the name of the object's next attribute.
+| [SetAttrValue](#function-setattrvalue)                    | Set the value of an object's attribute.
+| [AddAttr](#function-addattr)                              | Add a new attribute to an object.
+| [OpenAttr](#function-openattr)                            | Open an attribute as if it were an object with content.
+| [GetFirstMethod](#function-getfirstmethod--getnextmethod) | Get the name of an object's first method.
+| [GetNextMethod](#function-getfirstmethod--getnextmethod)  | Get the name of an object's next method.
+| [ExecuteMethod](#function-executemethod)                  | Execute a method with a given name and optional parameter string.
+| [PresentationHints](#function-presentationhints)          | Get info about an object's attributes.
+| [Info](#function-info)*                                   | Get info about an object instance.
+| [Commit](#function-commit)                                | Commit changes made to an object.
+| [GetQueryCoverageMask](#function-getquerycoveragemask)    | ???
+| [GetQueryIdentityPath](#function-getqueryidentitypath)    | ???
+
+_*Function is always required._
+
+_**Function is required to support queries._
```

Comment:
Those must be implemented but can return NULL if the driver does not support queries for some reason.

Reply:
Yeah, the rootnode driver is a special case with very limited functionality.